meeko/flexibility.py

```diff
@@ -5,275 +5,269 @@
+    compensation = -1  # first atom after rotatable bond doesn't move, so doesn't weigh
+    # atoms in breakable bonds count twice, because of added complexity
+    compensation += int(sum([i in glue_atoms for i in members[seed_node]]))
+    this_value = depth**2 * (nr_atoms + compensation)
+    max_value = this_value
+    visited.append(seed_node)
+    for node in graph[seed_node]:
+        if node not in visited:
+            visited.append(node)
+            new_value = _calc_max_weighted_depth(model, node, bonds_to_break, visited, depth + 1)
+            max_value = max(max_value, new_value + this_value)
+    return max_value
+
+
+def merge_terminal_atoms(flex_model, not_terminal_atoms=()):
+    members_dict = flex_model["rigid_body_members"]
+    graph = flex_model["rigid_body_graph"]
+    remove = {}
+    for group_index, members in members_dict.items():
+        if (
+            len(members) == 1
+            and len(graph[group_index]) == 1
+            and members[0] not in not_terminal_atoms
+            and group_index != flex_model["root"]
+        ):
+            remove[group_index] = members[0]
+    flex_model["rigid_body_count"] -= len(remove)
+    for group_index, atom_index in remove.items():
+        host_index = graph[group_index][0]
+        flex_model["rigid_body_members"].pop(group_index)
+        flex_model["rigid_body_members"][host_index].append(atom_index)
+        flex_model["rigid_body_graph"].pop(group_index)
+        flex_model["rigid_body_graph"][host_index].remove(group_index)
+        flex_model["rigid_body_connectivity"].pop((host_index, group_index))
+        flex_model["rigid_body_connectivity"].pop((group_index, host_index))
+        flex_model["rigid_index_by_atom"][atom_index] = host_index
+    return
+
+
+def get_flexibility_model(
+    molsetup,
+    root_atom_index=None,
+    break_combo_data=None,
+    glue_pseudo_atoms=None,
+):
+    # No macrocyclic rings are to be broken. We simply build the flexibility model.
+    if break_combo_data is None or len(break_combo_data["bond_break_combos"]) == 0:
+        bonds_to_break = ()
+        unbroken_rings_bonds = []
+        for ring in molsetup.rings:
+            for bond in molsetup.get_bonds_in_ring(ring):
+                unbroken_rings_bonds.append(bond)
+        flex_model = walk_rigid_body_graph(molsetup, bonds_to_break, unbroken_rings_bonds)
+        nr_not_ignored = sum(
+            [not molsetup.atom_ignore[i] for i in range(len(molsetup.atom_ignore))]
+        )
+        if len(flex_model["visited"]) != nr_not_ignored:
+            molsetup.show()
+            for key, value in flex_model.items():
+                print(key, value)
+            msg = f"{len(flex_model['visited'])=} differs from not-ignored atoms {nr_not_ignored}"
+            raise RuntimeError(msg)
+        root_body_index = get_root_body_index(flex_model, root_atom_index)
+        flex_model["root"] = root_body_index
+        broken_bonds = []
```

Comment:
Is there a reason we're bothering to return broken_bonds if it's an empty list always anyway?

Reply:
that's the first return statement of this function, the second one returns non-empty lists: https://github.com/forlilab/Meeko/blob/6317966790b4e9817e37649bf771adb13bc3d921/meeko/flexibility.py#L126